com.unity.render-pipelines.high-definition/Runtime/PostProcessing/PostProcessSystem.RenderGraph.cs

```diff
@@ -65,113 +188,235 @@ class AlphaCopyPassData
-                //                        }
-                //                    }
-                //                }
+                    source = passData.source;
+                }
+
+                // Optional NaN killer before post-processing kicks in
+                bool stopNaNs = hdCamera.stopNaNs && m_StopNaNFS;
+
+#if UNITY_EDITOR
+                if (isSceneView)
+                    stopNaNs = HDAdditionalSceneViewSettings.sceneViewStopNaNs;
+#endif
+                if (stopNaNs)
+                {
+                    using (var builder = renderGraph.AddRenderPass<StopNaNPassData>("Stop NaNs", out var passData, ProfilingSampler.Get(HDProfileId.StopNaNs)))
+                    {
+                        passData.source = builder.ReadTexture(source);
+                        passData.parameters = PrepareStopNaNParameters(hdCamera);
+                        TextureHandle dest = renderGraph.CreateTexture(new TextureDesc(Vector2.one, true, true)
+                        {
+                            name = "Stop NaNs Destination",
+                            colorFormat = m_ColorFormat,
+                            useMipMap = false,
+                            enableRandomWrite = true
+                        });
+                        passData.destination = builder.WriteTexture(dest); ;
+
+                        builder.SetRenderFunc(
+                        (StopNaNPassData data, RenderGraphContext ctx) =>
+                        {
+                            DoStopNaNs(data.parameters, ctx.cmd, ctx.resources.GetTexture(data.source), ctx.resources.GetTexture(data.destination));
+                        });
+
+                        source = passData.destination;
+                    }
+                }
+
+                // Dynamic exposure - will be applied in the next frame
+                // Not considered as a post-process so it's not affected by its enabled state
+                // Dynamic exposure - will be applied in the next frame
+                // Not considered as a post-process so it's not affected by its enabled state
+                if (!IsExposureFixed(hdCamera) && m_ExposureControlFS)
+                {
+                    var exposureParameters = PrepareExposureParameters(hdCamera);
+
+                    GrabExposureRequiredTextures(hdCamera, out var prevExposure, out var nextExposure);
+
+                    var prevExposureHandle = renderGraph.ImportTexture(prevExposure);
+                    var nextExposureHandle = renderGraph.ImportTexture(nextExposure);
+
+                    using (var builder = renderGraph.AddRenderPass<DynamicExposureData>("Dynamic Exposure", out var passData, ProfilingSampler.Get(HDProfileId.DynamicExposure)))
+                    {
+                        passData.source = builder.ReadTexture(source);
+                        passData.parameters = PrepareExposureParameters(hdCamera);
+                        passData.prevExposure = prevExposureHandle;
```

Comment:
An explicit Write/Read should still be done here.
The reason is that the pass needs to know that it writes to an imported texture so that it's considered as side effect and cannot be pruned.